plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -104,15 +106,21 @@ export class SonataFlowService {
         ? `${args.serviceUrl}/${args.definitionId}?businessKey=${args.businessKey}`
         : `${args.serviceUrl}/${args.definitionId}`;
 
-      const result = await fetch(urlToFetch, {
+      const response = await fetch(urlToFetch, {
         method: 'POST',
         body: JSON.stringify(args.inputData),
         headers: { 'content-type': 'application/json' },
       });
 
-      const json = await result.json();
-      this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
-      return json;
+      if (response.ok) {
+        const json = await response.json();
+        this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
+        return json;
+      }
+      const res = await response.json();
+      throw new Error(
+        `${this.createPrefixFetchErrorMessage(urlToFetch, 'executeWorkflow', 'POST')} - Details: ${res?.details}, Stack: ${res?.stack}`,
```

Comment:
I think many times you won't have the stack and the details and the error message will not be user friendly

Reply:
Also better separate the whole generation of error message, we will need it for other requests as well

Reply:
Also I'm not sure you can access details directly. You might need to do res.json().details
Is there a reproduction?

Reply:
@batzionb

> I think many times you won't have the stack and the details and the error message will not be user friendly

please see this discussion https://github.com/janus-idp/backstage-plugins/pull/2169#discussion_r1768742981

> Also better separate the whole generation of error message, we will need it for other requests as well

I'll adress this

> Also I'm not sure you can access details directly. You might need to do res.json().details
Is there a reproduction?

Can you clarify? 
First I've this `const res = await response.json();` and then ` Details: ${res?.details}`, what's wrong?
I verified it both with unit test and on the cluster, I didn't see any issue.

Reply:
it works on the cluster with res?.details

Reply:
if that's the case then no need for 
const res = await response.json();

Reply:
sorry didn't notice you had
const res = await response.json();
so comment irrelevant

Reply:
Sorry I missed that you already have
const res = await response.json();
You can ignore this comment